src/include/miopen/interpolate/problem_description.hpp

```diff
@@ -0,0 +1,256 @@
+
+#pragma once
+
+#include "miopen/miopen.h"
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace interpolate {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const TensorDescriptor& scaleFactorsDesc_,
+                       const miopenInterpolateMode_t mode_,
+                       const bool align_corners_,
+                       bool is_fwd_)
+        : scaleFactorsDesc(scaleFactorsDesc_),
+          mode(mode_),
+          align_corners(align_corners_),
+          is_fwd(is_fwd_)
+    {
+        IsValidMode();
+    }
+
+    const TensorDescriptor& GetScaleFactorsDesc() const { return scaleFactorsDesc; }
+    miopenInterpolateMode_t GetMode() const { return mode; }
+    bool GetAlignCorners() const { return align_corners; }
+
+    bool IsValidMode() const
+    {
+        if(mode != MIOPEN_INTERPOLATE_MODE_NEAREST && mode != MIOPEN_INTERPOLATE_MODE_LINEAR &&
+           mode != MIOPEN_INTERPOLATE_MODE_BILINEAR && mode != MIOPEN_INTERPOLATE_MODE_TRILINEAR &&
+           mode != MIOPEN_INTERPOLATE_MODE_BICUBIC)
+        {
+            std::cout << "MODE: " << mode << std::endl;
+            MIOPEN_THROW(miopenStatusBadParm, "Interpolate: Invalid mode.");
+        }
+        return true;
+    }
+
+protected:
+    TensorDescriptor scaleFactorsDesc;
+    miopenInterpolateMode_t mode;
+    bool align_corners = false;
+    bool is_fwd;
+
+    NetworkConfig MakeForwardNetworkConfig() const;
+};
+
+struct FwdProblemDescription : ProblemDescription
+{
+    FwdProblemDescription(const TensorDescriptor& inputDesc_,
+                          const TensorDescriptor& outputDesc_,
+                          const TensorDescriptor& scaleFactorsDesc_,
+                          const miopenInterpolateMode_t mode_,
+                          const bool align_corners_)
+        : ProblemDescription(scaleFactorsDesc_, mode_, align_corners_, true)
+    {
+        inputDesc  = inputDesc_;
+        outputDesc = outputDesc_;
+        IsValidDims();
+        IsValidLength();
+    }
+
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+
+    bool IsValidLength() const
+    {
+        if(inputDesc.GetSize() < 3 || inputDesc.GetSize() > 5)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "Interpolate: Input tensor size < 3 or > 5 is not valid.");
+        }
+
+        if(outputDesc.GetSize() < 3 || outputDesc.GetSize() > 5)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "Interpolate: Output tensor size < 3 or > 5 is not valid.");
+        }
+
+        if((outputDesc.GetSize() - 2) != scaleFactorsDesc.GetElementSize())
+        {
+            if(mode != MIOPEN_INTERPOLATE_MODE_NEAREST)
```

Comment:
May I know why do you have to exclude nearest mode in this check?

Reply:
Cuz I use only 5d tensor in nearest kernel as Modnn, it leads to always 3d length scaleFactor no matter what the dims of output or input are

Reply:
If it is like you said, i think you should also check scaleFactorsDesc.GetElementSize() == 3 when mode is nearest

Reply:
Okay, I always extend the scale factor to a length of 3 in the driver when it is in nearest mode, so you can use the driver in that mode as usual.

Reply:
The scale factor of this mode need always be padded properly 1 in layer above miopen_api layer as I code in driver, so I think this is redundant